File: ClientApp/src/app/models/workTask.ts
import { Employee } from "./employee";
import { Project } from "./project";
import { TaskState } from "./taskState";
import { TaskStateRecord } from "./taskStateRecord";

export class WorkTask
{

    id:number=0;
    title:string="";
    description:string="";
    dueDate:Date=new Date();
    state:TaskState=TaskState.ToDo;
    assignee:Employee=new Employee();
    states:Array<TaskStateRecord>=new Array<TaskStateRecord>();
    project:Project=new Project();
    projectId:number=0;
}

File: ServerApp/Model/WorkTask.cs
﻿
namespace Model;

public class WorkTask
{
    public string Title { get; set; } = "";

    public string Description { get; set; } = "";
    public virtual Employee? Assignee { get; set; }
    public long? AssigneeId { get; set; }
    public DateTime DueDate { get; set; }
    public long Id { get; set; }
    public List<TaskStateRecord> States { get; set; } = new();

    public Project? Project { get; set; } 
    public long ProjectId { get; set; } 

}
